designer/server/src/plugins/routes/api.test.js

```diff
@@ -121,18 +109,22 @@ describe('Server API', () => {
       }
     }
 
-    const result = await server.inject<{ err: Error }>(options)
-    expect(result.statusCode).toBe(401)
+    const result =
+      /** @type {import('@hapi/hapi').ServerInjectResponse<{ err: Error }>}) */ (
+        await server.inject(options)
+      )
```

Comment:
Can use the shorthand for these if you like?

```suggestion
    const result = /** @type {ServerInjectResponse<{ err: Error }>}) */ (
      await server.inject(options)
    )
```

<br>

Until we get `@import` support you'll need to pair `@jsdoc` with `@template`

```js
/**
 * @template {object} Result
 * @typedef {import('@hapi/hapi').ServerInjectResponse<Result>} ServerInjectResponse
 */
```